todo-list/script.js

```diff
@@ -1,150 +1,231 @@
-        todos = [
-            ...todos,
-            {
-                label: this.value,
-                checked
-            }
-        ];
-        this.value = '';
-        renderTodos();
+class Task {
+    constructor(label) {
+        this.label = label;
+        this.isCompeleted = false;
     }
 }
 
-// Handler checked todo task
-function handlerCheckedTodo(event) {
-    if(event.target.tagName === 'INPUT' && event.target.getAttribute('name') == 'todo-task') {
-        const id = parseInt(event.target.closest('li').getAttribute('id'), 10);
-        todos[id].checked = !todos[id].checked;
-        renderTodos();
+class TodoList {
+    
+    constructor(elementInitApp, preset = {}) {
+        this.titleOfApp = preset.title ? preset.title : 'Todo List App';
+        this.idOfApp = preset.id ? preset.id : '';
+        this.elementInitApp = elementInitApp;
+        this.state = [];
+        this.createStructureApp();
+        this.checkDataForState();
     }
-}
 
-// Delete Todo item
-function handlerDeleteTodo(event) {
-    if(event.target.getAttribute('class') === 'delete-todo') {
-        const id = parseInt(event.target.closest('li').getAttribute('id'), 10);
-        if(confirm(`Are you sure to delete this task ==> ${todos[id].label}`)) {
-            todos = todos.filter(todoObj => todoObj.label != todos[id].label);
-            renderTodos();
+    checkDataForState() {
+        const getDataFromlocalStorage = localStorage.getItem(this.elementInitApp.getAttribute('id'));
+        if(getDataFromlocalStorage) {
+            this.state = JSON.parse(getDataFromlocalStorage);
+            this.renderTodoTasks();
+        } else {
+            this.state = [];
         }
     }
-}
 
-// Edite Todo Item
-function handlerEditTodo(event) {
-    if(event.target.getAttribute('class') === 'edit-todo') {
-        // wrapperList.removeEventListener('click', handlerCheckedTodo);
-        const id = parseInt(event.target.closest('li').getAttribute('id'), 10);
-        const todoText = app.querySelector(`li[id='${id}'] .todo-text`);
-        const todoEdit = app.querySelector(`li[id='${id}'] .todo-edit`);
-
-        // hidden true todo text
-        todoText.hidden = true;
-        // hidden false input edite
-        todoEdit.hidden = false;
-
-        // replace this value input to todo list array
-        function changeValueOfTodo(event) {
-            todos[id].label = this.value;
-            renderTodos();
-
-            // Clean the memory
-            todoEdit.hidden = true;
-            todoEdit.removeEventListener('change', changeValueOfTodo);
-        }
-        todoEdit.addEventListener('change', changeValueOfTodo);
+    // helper method
+    createElement(elementName, value='', addClass='', id='') {
+        const element =  document.createElement(elementName);
+        if(addClass) element.classList.add(addClass);
+        if(value) element.innerText = value;
+        if(id) element.setAttribute('id', id + '');
```

Comment:
Why `id + '' `

Reply:
because sometimes not set id to element and i think it should again convert to string